server/game/cards/08.6-SAT/GourdinDeBoisFlotte.js

```diff
@@ -0,0 +1,33 @@
+const DrawCard = require('../../drawcard.js');
+
+class GourdinDeBoisFlotte extends DrawCard {
+    setupCardAbilities(ability) {
+        this.attachmentRestriction({ trait: 'Drowned God' });
+
+        this.whileAttached({
+            effect: ability.effects.dynamicStrength(() => this.controller.deadPile.length)
+        });
+
+        this.interrupt({
+            when: {
+                onCharacterKilled: event => event.cardStateWhenKilled === this.parent
```

Comment:
This won't work as cardStateWhenKilled is not a perfect copy of the original card object.

But, since this is an interrupt, simply using `onCharacterKilled: event => event.card === this.parent` works.